### PR TITLE
Docs: Bigtable + Dataflow hello world cleanups

### PR DESCRIPTION
**Pull request by billyjacobson**
- [ ] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [ ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: bigtable/beam/helloworld/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    ```cbt createtable mobile-time-series families="stats_summary"```
+
+1. Set up the environment variables
+
+    ```
+    GOOGLE_CLOUD_PROJECT=your-project-id
+    BIGTABLE_PROJECT=your-project-id
+    INSTANCE_ID=your-instance-id
+    TABLE_ID=your-instance-id
```
Comment (kolea2) [inline bigtable/beam/helloworld/README.md:20]: "your-table-id"*

File: bigtable/beam/helloworld/pom.xml [hunk 2]
```diff
@@ -37,7 +37,10 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <maven.compiler.source>1.8</maven.compiler.source>
     <maven.compiler.target>1.8</maven.compiler.target>
-    <apache_beam.version>2.31.0</apache_beam.version>
+    <!-- Keep this apache_beam.version in sync with version used in bigtable-hbase-beam -->
```
Comment (kolea2) [inline bigtable/beam/helloworld/pom.xml:40]: Let's add an exclusion to the[ renovate config](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/renovate.json) so this doesn't accidentally get updated in the future.

File: bigtable/beam/helloworld/pom.xml [hunk 3]
```diff
@@ -37,7 +37,10 @@
...
+    <!-- https://mvnrepository.com/artifact/com.google.cloud.bigtable/bigtable-hbase-beam -->
+    <apache_beam.version>2.24.0</apache_beam.version>
+    <exec.cleanupDaemonThreads>false</exec.cleanupDaemonThreads>
```
Comment (kolea2) [inline bigtable/beam/helloworld/pom.xml:43]: what's this for?

File: bigtable/beam/helloworld/pom.xml [hunk 4]
```diff
@@ -76,6 +80,18 @@
...
       <version>1.1.3</version>
       <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>com.google.guava</groupId>
+          <artifactId>guava</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <!-- guava is required override the guava from the shared-configuration which causes build issues -->
```
Comment (kolea2) [inline bigtable/beam/helloworld/pom.xml:90]: I would reword this slightly:

"manually declare guava to override version from shared-configuration"

File: bigtable/beam/helloworld/src/test/java/HelloWorldTest.java [hunk 5]
```diff
@@ -98,13 +100,21 @@ public static void afterClass() {
     }
   }
 
+  /**
+   * Test the writing data example as well as runtime dependency building. This must be run via
```
Comment (kolea2) [inline bigtable/beam/helloworld/src/test/java/HelloWorldTest.java:104]: I would reword slightly:

"This is tested via dataflow to ensure there are no dependency conflicts which can cause runtime errors."

File: bigtable/beam/helloworld/pom.xml (hunk 3 above)
Comment (billyjacobson) [reply bigtable/beam/helloworld/pom.xml:43]: When you run it locally it prevents a bunch of warnings about threads

**Review by kolea2 — APPROVED** [review]
